ambry-account/src/main/java/com/github/ambry/account/HelixAccountService.java

```diff
@@ -263,7 +263,7 @@ public boolean updateAccounts(Collection<Account> accounts) {
    * @return True when the update operation succeeds.
    */
   boolean updateAccountsWithAccountMetadataStore(Collection<Account> accounts,
-      AccountMetadataStore accountMetadataStore) {
+      AccountMetadataStore accountMetadataStore) throws AccountServiceException {
```

Comment:
It seems this method uses both `return false` and AccountServiceException to handle failure cases. Can we throw AccountServiceException everywhere to make it consistent? Also we can consider removing boolean return value, if the method goes through without exception, then it should be a success.

Reply:
Feel free to add new type of error code to `AccountServiceErrorCode`

Reply:
I didn't know what to do about the case where config.updateDisabled = true.  What exception should that be?  Do we ever actually turn off account updates?

Reply:
Maybe this case justifies `TemporaryDisabled` and from customer point's of view, it's 503 (Unavailable) error. 
This config was introduced to migrate ZK based metadata store to router based metadata store. It gives us an option to temporarily turn off any account update during the migration (to avoid potential race condition). 
We can remove this or if you think it's useful for account service migration, then we can keep it.